core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -873,6 +875,39 @@ object Chunk {
       case _: BitChunk       => ClassTag.Boolean.asInstanceOf[ClassTag[A]]
     }
 
+  /**
+   * The maximum number of elements in the buffer for fast append.
+   */
+  private val BufferSize: Int =
```

Comment:
This one is going to be tricky. I'm not sure the right number to choose. I'd think somewhere between `32` and `128`. Higher numbers equal faster performance for repeated adds, while lower numbers yield higher waste for infrequent adds.

Reply:
Maybe we start with 64 and we can always adjust? I think we probably need some aggregate benchmark for a varied set of tasks to really pick the "right" number here and may also make sense to get in some of the other improvements we're making before doing that degree of optimization.

Reply:
Makes sense to me! 👍

Reply:
One other thing we could do here is have some heuristic for the buffer size. Like we could start out with a buffer size of `1` and double the buffer size every time something is "too big" so we are learning and optimizing for how the user is working with the chunk.

Reply:
My feeling is that it will not perform that well compared to just choosing a static number.

I remember some research showing the vast majority of collections are "small collections". The constant doubling (from 1 to 2, 2 to 4, etc.) means a lot of copying on your way to a small size.

Reply:
Yes. We could also start at an intermediate size like say `8` and double from there but also fine as is and can always optimize later.